src/App.tsx

```diff
@@ -50,9 +57,17 @@ function App() {
   };
 
   const handleUpload = async (files: File[]) => {
+    let converter: MediaConverter | undefined;
+    console.log(files[0].type);
     if (isImageFile(files[0])) {
+      converter = imageConverter;
+    } else if (isFontFile(files[0])) {
+      converter = fontConverter;
+    }
+    console.log(converter);
+    if (converter !== undefined) {
       toast.promise(
-        imageConverter.convert(
+        converter.convert(
           files.map((file: File) => ({ filepath: file.name, data: file }))
```

Comment:
This might be irrelevant at the moment, but when doing the POST call, the file key should be the path relative to the user's game directory that they specified. Example:

```
bundle.zip
| game
|-- main.lua
|-- graphics
|---- test.png
```

The POSTed file path should be `graphics/test.png` and the server will respond in the JSON with the filepath`graphics/test.t3x`. This way you can write the decoded base64 data to the proper location in the in-memory zip file.

Reply:
It will have to be accounted for when doing the zip compilation, by passing the full path to `filepath`. The `convert` method takes in a `MediaFile` type, which is an alias for `{ data: Blob; filepath: string; }`, and in the POST the `filepath` is used as the key.